sdk/storage/blobs/inc/az_storage_blobs.h

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STORAGE_BLOBS_H
+#define AZ_STORAGE_BLOBS_H
+
+#include <az_contract.h>
+#include <az_http_pipeline.h>
+#include <az_http_response.h>
+#include <az_result.h>
+#include <az_span.h>
+#include <az_str.h>
+
+#include <stddef.h>
+
+#include <_az_cfg_prefix.h>
+
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_header_blob_type() { return AZ_STR("x-ms-blob-type"); }
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_header_Content_Length() { return AZ_STR("Content_Length"); }
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_AppendBlob() { return AZ_STR("AppendBlob");} /* UnSupported */
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_BlockBlob() { return AZ_STR("BlockBlob");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_blob_type_PageBlob() { return AZ_STR("PageBlob"); } /* UnSupported */
+
+typedef struct {
+  az_span service_version;
+  az_http_policy_retry_options retry;
+} az_storage_blobs_blob_client_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_upload_options;
+
+typedef struct {
+  az_span option;
+} az_storage_blobs_blob_download_options;
+
+typedef struct {
+  az_span uri;
+  az_span blob_type;
```

Comment:
The blob type wouldn't be a field in the client.
If I just give you a URL, you can't tell what kind of blob it is.
In the other languages, we have block blob client, an append blob client, and a page blob client. But we also have a blob client (for unknown blob types) as a base class for the others.
You download from all 3 kinds of blobs the same way. The difference between the 3 types is how you upload. I think we'll only need to support block blobs in the C SDK.

Reply:
C doesn't give us the luxury of base classes.  Thus I suggest this client becomes the az_blobs_block_blob_client.

That way if we are future proofed.  We can add other types of blobs in the future.